vault/identity_store_util.go

```diff
@@ -332,7 +332,7 @@ func (i *IdentityStore) upsertEntityInTxn(ctx context.Context, txn *memdb.Txn, e
 			// Otherwise it's still tied to previousEntity and fall through
 			// into merging
 			fallthrough
```

Comment:
This fallthrough will blow by the check you put in since it doesn't consider the case statement.

Reply:
`fallthrough` only indicates not to break from the switch and to follow on to check the subsequent case statements. It is not a by-pass command for the case. So, I think this doesn't harm.

Reply:
Fallthrough does not consider case statements. See https://github.com/golang/go/wiki/Switch#fall-through.